java/java.lsp.server/vscode/src/extension.ts

```diff
@@ -41,7 +41,7 @@ import * as launcher from './nbcode';
 import { StatusMessageRequest, ShowStatusMessageParams, QuickPickRequest, InputBoxRequest } from './protocol';
 
 const API_VERSION : string = "1.0";
-let client: Promise<LanguageClient>;
+let client: Promise<LanguageClient> | null = null;
```

Comment:
Ideally this `client` promise could be set soon to be defined (not `undefined`) and resolved later. That would mean to make sure the `client` promise is set sooner then `registerCommand` is registered. E.g. 
```
let c : LanguageClient = await client
```
can only be called when the `client` is defined.

Reply:
Agreed. Fixed.